src/cosalib/qemuvariants.py

```diff
@@ -204,6 +206,14 @@ def mutate_image(self, callback=None):
         cmd.extend([work_img])
         run_verbose(cmd)
 
+        img_info = image_info(work_img)
+        if self.image_format != img_info.get("format"):
+            raise ImageError((f"{work_img} format mismatch"
+                              f" expected: '{self.image_format}'"
+                              f" found: '{img_info.get('format')}'"))
+
+        if not callback:
```

Comment:
So right now, an implementing class has three ways to add a callback, right? Using `self.mutate_callback =`, or `mutate-callback`, or setting the `callback` arg? Can we reduce all this to strictly just the `self.mutate_callback` approach used here to make it easier to grok?

Reply:
/me nods

I thought that was a little wonky. Consolidated it down to a constructor argument by default.